### PR TITLE
chore: flagd add offline flag source path support through env variables

### LINKED ISSUE #646: [flagd] offline mode file path from env variable
Similar to go-sdk provider, we should allow to configure java provider's offline flag source path from an environment variable.

For consistency use same variable name - `FLAGD_OFFLINE_FLAG_SOURCE_PATH`

**AC**

Env variable is available and documentation updated accordingly 

See - _Originally posted by @Kavindu-Dodan in https://github.com/open-feature/go-sdk-contrib/pull/421#discussion_r1463811416_

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

Fixes #646 by adding `FLAGD_OFFLINE_FLAG_SOURCE_PATH` env variable to set offline mode's flag source path.

Logic is improved to handle null and documentation was updated to have consistency across languages.

### CODE REVIEW

**Review by beeme1mr — APPROVED** [review]

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdOptions.java [hunk 1]
```diff
@@ -135,31 +130,14 @@ public class FlagdOptions {
...
-        }
-
         /**
          * Enable OpenTelemetry instance extraction from GlobalOpenTelemetry. Note that, this is only useful if global
          * configurations are registered.
          */
         public FlagdOptionsBuilder withGlobalTelemetry(final boolean b) {
-            this.openTelemetry = GlobalOpenTelemetry.get();
-
+            if (b) {
```
Comment (thisthat) [inline providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdOptions.java:138]: 🤕 how did we miss seeing this? nice catch :shipit:

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java [hunk 2]
```diff
@@ -46,9 +46,10 @@ public class InProcessResolver implements Resolver {
      * Initialize an in-process resolver.
      */
     public InProcessResolver(FlagdOptions options, Consumer<ProviderState> stateConsumer) {
-        final Connector connector = options.isOffline()
-                ? new FileConnector(options.getOfflineFlagSourcePath())
-                : new GrpcStreamConnector(options);
+        final Connector connector =
```
Comment (thisthat) [inline providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java]: maybe we could add some tests in InProcessResolverTest.java to check that the right connector is selected but def not needed as part of this PR :)

**Review by thisthat — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdOptions.java (hunk 1 above)
Comment (Kavindu-Dodan) [reply providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdOptions.java:138]: I saw this when cleaning the builder logic 😿

File: providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java (hunk 2 above)
Comment (Kavindu-Dodan) [reply providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java]: Done, added a test with commit https://github.com/open-feature/java-sdk-contrib/pull/647/commits/f2c331338e96047723f49d3cea9e809301e1d6d4

**Pull request merged**